ext/http/http_next.rs

```diff
@@ -231,20 +232,56 @@ where
     &request_parts.headers,
   );
 
+  let method: v8::Local<v8::Value> = v8::String::new_from_utf8(
+    scope,
+    request_parts.method.as_str().as_bytes(),
+    v8::NewStringType::Normal,
+  )
+  .unwrap()
+  .into();
+
+  let authority: v8::Local<v8::Value> = match request_properties.authority {
+    Some(authority) => v8::String::new_from_utf8(
+      scope,
+      authority.as_ref(),
+      v8::NewStringType::Normal,
+    )
+    .unwrap()
+    .into(),
+    None => v8::undefined(scope).into(),
+  };
+
   // Only extract the path part - we handle authority elsewhere
   let path = match &request_parts.uri.path_and_query() {
     Some(path_and_query) => path_and_query.to_string(),
     None => "".to_owned(),
   };
 
-  // TODO(mmastrac): Passing method can be optimized
-  (
-    request_parts.method.as_str().to_owned(),
-    request_properties.authority,
-    path,
-    String::from(request_info.peer_address.as_ref()),
-    request_info.peer_port,
+  let path: v8::Local<v8::Value> =
+    v8::String::new_from_utf8(scope, path.as_ref(), v8::NewStringType::Normal)
+      .unwrap()
+      .into();
+
+  let peer_address: v8::Local<v8::Value> = v8::String::new_from_utf8(
+    scope,
+    request_info.peer_address.as_bytes(),
+    v8::NewStringType::Normal,
   )
+  .unwrap()
+  .into();
+
+  let port: v8::Local<v8::Value> = match request_info.peer_port {
+    Some(port) => v8::Integer::new(scope, port.into()).into(),
+    None => v8::undefined(scope).into(),
+  };
+
+  let vec: SmallVec<[v8::Local<v8::Value>; 5]> =
```

Comment:
As we know the exact size beforehand, can we somehow create `v8::Array` directly, skipping `SmallVec` part here?

Reply:
In this case, I recommend creating a Rust array of 5 elements instead of using `SmallVec`.

Reply:
Looking at the docs, this should also do the trick I think?

```rs
let array = v8::Array::new(scope, 5);
array.set_index(scope, 0, method);
array.set_index(scope, 1, authority);
array.set_index(scope, 2, path);
array.set_index(scope, 3, peer_address);
array.set_index(scope, 4, port);
```

Reply:
I ended up using a simple array, as it's fixed-size anyway so we don't need vec here either.